recipe/meta.yaml

```
@@ -10,36 +10,36 @@ package:
 
 source:
   url: https://github.com/NVIDIA/cuda-python/archive/refs/tags/v{{ version }}.tar.gz
-  sha256: ef54ada5d607387f59950cfb3c3852e5816b62824402d289227b8066e8353489
+  sha256: bfb36d314065b695decd23935ab008630cc956cbcd428d520106568fdeb7e644
 
 build:
-  number: 2
+  number: 0
   skip: true  # [py<37 or osx or cuda_compiler_version != "11.8"]
   script:
     - {{ PYTHON }} -m pip install . --no-deps -vv
   run_exports:
     - cuda-python >=11.7.1,<12  # we need this because CUDA Python 11.7.1 has ABI breaks, see conda-forge/cuda-python-feedstock#15
-  ignore_run_exports:
-    - cudatoolkit
+  ignore_run_exports_from:
+    - {{ compiler('cuda') }}
 
 requirements:
   build:
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
     - {{ compiler('cuda') }}
-    - sysroot_linux-64 2.17  # [linux64]
+    - sysroot_{{ target_platform }} 2.17  # [linux]
   host:
     - setuptools
     - cython
     - python
     - pyclibrary
     - pip
+    - cuda-version {{ major_version }}
   run:
     - python
-    - {{ pin_compatible('cudatoolkit', min_pin='x', max_pin='x') }}
+    - {{ pin_compatible('cuda-version', min_pin='x', max_pin='x') }}
+    - cudatoolkit  # [(cuda_compiler_version or "").startswith("11")]
```

Here's the alternative using Jinja

```suggestion
    {% if major < 12 %}
    - cudatoolkit
    {% endif %}
```

---

I'm happy with either approach as well. I think that if the package lists for CUDA 12 are more than a couple lines, I slightly prefer Jinja if-blocks over selectors on each line, just to avoid repeating the same selectors over and over.

---

+1 on the jinja if blocks

---

Related question, since this is a run dependency, if someone has cuda-* 12.0 installed in their environment, what would happen if they tried to install cuda-python 11.8 (either directly or because a package depends on it)? Wouldn't this force the environment to resolve to cudatoolkit rather than cuda-*>=12 (which would also work)?

---

`cuda-version` would cause a conflict here, so to install 11.8 packages to an already-12.0 env will force conda to resolve the conflict, I think in this case it'd force the env to downgrade to 11.8.

---

(basically, should this be a run or a run_constrained?)

---

Since `cudatoolkit` was in `run`, this should be too

---

Anyways sounds like there is a preference for Jinja. Will go ahead and commit this. Happy to discuss anything else as needed

---

cudatoolkit was in run because we did not have cuda-* packages. Now the case is that *either* may be used to satisfy cuda-python. 

The ask here is how to actually reflect that or if there is no way to reflect that.